tests/ui/impl-trait/issues/issue-70877.rs

```diff
@@ -27,7 +27,9 @@ fn ham() -> Foo {
 
 fn oof(_: Foo) -> impl std::fmt::Debug {
     let mut bar = ham();
-    let func = bar.next().unwrap();
+    // Need to UFC invoke `Iterator::next`,
```

Comment:
I'm a bit surprised by this. I expect this to be equivalent to `let mut bar: Option<_> = None; bar.unwrap().next()` -- i.e., an error, yes, but not for this reason?